_posts/2018-02-13-the-pros-and-cons-of-state-management.md

```diff
@@ -0,0 +1,66 @@
+---
+layout: post
+title: "The Pros and Cons of State Management"
+date: 2018-02-13 09:11:01
+tags:
+- Web
+- Vue
+- Vuex
+- JavaScript
+- StateManagement
+twitter_text: "The Pros and Cons of #StateManagement"
+authors: Jaime Jones
+---
+
+Let's talk about state management for a minute. There's recently been a bunch of hubub about Redux on Twitter, and arguments about the usefulness of state management. Here at Ritter we haven't utilized Redux specifically, but we frequently leverage Vue's own state management library, Vuex.
+
+I'm going to do a brief overview of Vuex and then talk about the pros and cons of state management. There's something that I will point out right now, and reiterate later: State management is a very useful paradigm, but it is _not_ intended for use in every single project.
+
+## What is Vuex?
+I'm going to do a quick summary of Vuex and how it works, but I recommend that you check out the [Vuex docs](https://vuex.vuejs.org/en/intro.html) if you are interested or just looking for a more in depth explanation.
+
+Vuex is a state management library, aimed at creating a centralized source that can drive the rest of your application. It is composed of three main parts: State, Mutations, and Actions. For the sake of this explanation, we will use an example of a blog which must fetch the posts to display.
+
+State houses the data that lives in your application, and can be referred to as your 'single source of truth.' No matter where in your application you access data that is housed in the state, it will remain the same across each application because it is coming from this single source.
```

Comment:
`across each application` -- is this referring to `across each usage` or something like that? Using a different word may help clarity.

Reply:
Ah, yup, `across each component`